File: frontend/src/types/blog.ts
export interface BlogCategory {
  id: number;
  name: string;
  slug: string;
  description: string;
  created_at: string;
  updated_at: string;
  is_active: boolean;
  blog_count?: number;
}

export interface BlogTag {
  id: number;
  name: string;
  slug: string;
  created_at: string;
  updated_at: string;
  blog_count?: number;
}

export interface BlogImage {
  id: number;
  image?: string;
  image_url?: string;
  alt_text: string;
  created_at: string;
  updated_at: string;
  is_active: boolean;
}

export interface BlogAuthor {
  id: number;
  name: string;
  mobile: string;
  email?: string;
}

export interface Blog {
  id: number;
  title: string;
  slug: string;
  author: BlogAuthor;
  featured_image?: BlogImage;
  images?: BlogImage[];
  content: string; // JSON string from Lexical
  category?: BlogCategory;
  tags: BlogTag[];
  is_highlighted: boolean;
  is_published: boolean;
  published_at?: string;
  views_count: number;
  created_at: string;
  updated_at: string;
  read_time: number; // in minutes
}

export interface BlogListResponse {
  count: number;
  next?: string;
  previous?: string;
  results: Blog[];
}

export interface BlogFormData {
  title: string;
  slug?: string;
  content: string;
  category_id?: number | null;
  featured_image_id?: number | null;
  image_ids?: number[];
  tag_ids: number[];
  is_highlighted: boolean;
  is_published: boolean;
}

export interface BlogFilters {
  category?: string;
  tag?: string;
  search?: string;
  highlighted?: boolean;
}

File: hsquare/lib/models/blog.dart
class Blog {
  final int id;
  final String title;
  final String slug;
  final String excerpt;
  final String content;
  final String? featuredImage;
  final BlogCategory? category;
  final List<BlogTag> tags;
  final String author;
  final DateTime publishedAt;
  final int viewsCount;
  final bool isPublished;
  final bool isHighlighted;

  Blog({
    required this.id,
    required this.title,
    required this.slug,
    required this.excerpt,
    required this.content,
    this.featuredImage,
    this.category,
    required this.tags,
    required this.author,
    required this.publishedAt,
    required this.viewsCount,
    required this.isPublished,
    required this.isHighlighted,
  });

  factory Blog.fromJson(Map<String, dynamic> json) {
    return Blog(
      id: json['id'],
      title: json['title'],
      slug: json['slug'],
      excerpt: json['excerpt'] ?? '',
      content: json['content'] ?? '',
      featuredImage: json['featured_image'],
      category: json['category'] != null
          ? BlogCategory.fromJson(json['category'])
          : null,
      tags: (json['tags'] as List?)
          ?.map((tag) => BlogTag.fromJson(tag))
          .toList() ?? [],
      author: json['author'] ?? 'Admin',
      publishedAt: DateTime.parse(json['published_at']),
      viewsCount: json['views_count'] ?? 0,
      isPublished: json['is_published'] ?? true,
      isHighlighted: json['is_highlighted'] ?? false,
    );
  }
}

class BlogCategory {
  final int id;
  final String name;
  final String slug;

  BlogCategory({
    required this.id,
    required this.name,
    required this.slug,
  });

  factory BlogCategory.fromJson(Map<String, dynamic> json) {
    return BlogCategory(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
    );
  }
}

class BlogTag {
  final int id;
  final String name;
  final String slug;

  BlogTag({
    required this.id,
    required this.name,
    required this.slug,
  });

  factory BlogTag.fromJson(Map<String, dynamic> json) {
    return BlogTag(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
    );
  }
}

